geomesa-process/src/main/scala/org/locationtech/geomesa/process/HashAttributeProcess.scala

```diff
@@ -0,0 +1,84 @@
+package org.locationtech.geomesa.process
+
+import com.google.common.hash.Hashing
+import org.geotools.data.DataUtilities
+import org.geotools.data.simple.SimpleFeatureCollection
+import org.geotools.feature.collection.DelegateSimpleFeatureIterator
+import org.geotools.feature.simple.{SimpleFeatureBuilder, SimpleFeatureTypeBuilder}
+import org.geotools.process.ProcessException
+import org.geotools.process.factory.{DescribeParameter, DescribeProcess, DescribeResult}
+import org.geotools.process.vector.VectorProcess
+
+trait HashAttribute {
+  import org.locationtech.geomesa.utils.geotools.Conversions._
+
+  import scala.collection.JavaConversions._
+
+  val hashFn = Hashing.goodFastHash(64)
+
+  def transformHash(hash: Int): AnyRef
+  def augmentSft(sft: SimpleFeatureTypeBuilder): Unit
+
+  @throws(classOf[ProcessException])
+  @DescribeResult(name = "result", description = "Output collection")
+  def execute(@DescribeParameter(name = "data", description = "Input features")
+              obsFeatures: SimpleFeatureCollection,
+              @DescribeParameter(name = "attribute", description = "The attribute to hash on")
+              attribute: String,
+              @DescribeParameter(name = "modulo", description = "The divisor")
+              modulo: Integer): SimpleFeatureCollection = {
+
+    val sft = obsFeatures.getSchema
+    val sftBuilder = new SimpleFeatureTypeBuilder()
+    sftBuilder.init(sft)
+    augmentSft(sftBuilder)
+    val targetSft = sftBuilder.buildFeatureType()
+    val featureBuilder = new SimpleFeatureBuilder(targetSft)
+
+    val results =
+      obsFeatures.features().map { sf =>
+        featureBuilder.reset()
+        featureBuilder.init(sf)
+        val attr = Option(sf.getAttribute(attribute).asInstanceOf[String]).getOrElse("")
```

Comment:
Maybe change to Option(sf.getAttribute(attribute)).map(_.toString).getOrElse("") so that it handles any type?
Could also look up by index instead of name to speed up a little.